### PR TITLE
Fix MenuButton item label disappearing when setting shortcut.

### LINKED ISSUE #25519: MenuButton Label dissapears when a shortcut is set.
**Godot version:**
Latest from: https://hugo.pro/projects/godot-builds/ (v3.1.beta.calinou.c21ca98)

**OS/device including version:**
Windows 10

**Issue description:**
MenuButton items lose the label when a shortcut is set.

**Steps to reproduce:**
1. Create new scene with a Control node as the root
2. Add MenuButton node
3. Add this code to the root node:
```
extends Control

func _ready():
	# Add items
	$MenuButton.get_popup().add_item("I'm a regular item")
	$MenuButton.get_popup().add_item("I am not")
	
	# Create the shortcut
	var shortcut = ShortCut.new()
	var inputeventkey = InputEventKey.new()
	inputeventkey.set_scancode(KEY_Q)
	inputeventkey.control = true
	shortcut.set_shortcut(inputeventkey)
	
	# Set shortcut
	$MenuButton.get_popup().set_item_shortcut(1, shortcut, true)
```

4. Run project and check how the label on the second menu item is gone.

**Minimal reproduction project:**
[MenuButtonItemBug.zip](https://github.com/godotengine/godot/files/2818183/MenuButtonItemBug.zip)

### PR DESCRIPTION
**Pull request by Awkor**
Fixes #25519

I've used the minimal reproduction project provided in the issue to test the fix.

### CODE REVIEW

File: scene/gui/popup_menu.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ Size2 PopupMenu::get_minimum_size() const {
 			size.width += check_w + hseparation;
 		}
 
-		String text = items[i].shortcut.is_valid() ? String(tr(items[i].shortcut->get_name())) : items[i].xl_text;
+		String text = items[i].shortcut.is_valid() ? items[i].xl_text + String(tr(items[i].shortcut->get_name())) : items[i].xl_text;
```
Comment (akien-mga) [inline scene/gui/popup_menu.cpp]: Seems fine, though you can maybe write it as:
```
String text = items[i].xl_text;
if (items[i].shortcut.is_valid()) {
	text += String(tr(items[i].shortcut->get_name()));
}
```

Comment (akien-mga) [reply scene/gui/popup_menu.cpp]: Also, how does it look like after this fix? Isn't there a need for a space between the `xl_text` and the shortcut name?

Comment (Awkor) [reply scene/gui/popup_menu.cpp]: > 
> 
> Seems fine, though you can maybe write it as:
> 
> ```c++
> String text = items[i].xl_text;
> if (items[i].shortcut.is_valid()) {
> 	text += String(tr(items[i].shortcut->get_name()));
> }
> ```

On further inspection `tr(items[i].shortcut->get_name())` is not needed, it always returns an empty string, with or without shortcut.

The actual drawing of the shortcut is done [here](https://github.com/godotengine/godot/blob/1794b95fd2812b8fdff99258b4a6dc25818b58c4/scene/gui/popup_menu.cpp#L520)

> 
> 
> Also, how does it look like after this fix? Isn't there a need for a space between the `xl_text` and the shortcut name?

Here's how it looks:

image

Comment (akien-mga) [reply scene/gui/popup_menu.cpp]: Nice! So I guess it should be amended to simply `String text = items[i].xl_text;`?

Comment (Awkor) [reply scene/gui/popup_menu.cpp]: Yes, exactly

File: scene/gui/popup_menu.cpp [hunk 2]
```diff
@@ -79,7 +79,8 @@ Size2 PopupMenu::get_minimum_size() const {
 			size.width += check_w + hseparation;
 		}
 
-		String text = items[i].shortcut.is_valid() ? items[i].xl_text + String(tr(items[i].shortcut->get_name())) : items[i].xl_text;
+		String text = items[i].xl_text;
```
Comment (akien-mga) [inline scene/gui/popup_menu.cpp]: Are you sure that the width of the potential shortcut string is still properly taken into account after this change?

**Comment by akien-mga** [comment]
Could you squash commits together?

File: scene/gui/popup_menu.cpp (hunk 2 above)
Comment (Awkor) [reply scene/gui/popup_menu.cpp]: The width of the shortcut is calculated here:
https://github.com/godotengine/godot/blob/d56bc942c194c67c02bc049d8a6518ce0f882324/scene/gui/popup_menu.cpp#L88-L93

It then gets added with the label text width here:
https://github.com/godotengine/godot/blob/d56bc942c194c67c02bc049d8a6518ce0f882324/scene/gui/popup_menu.cpp#L104

But to be 100% sure i wrote the following test:
```GDScript
extends Control

onready var popup : PopupMenu = $MenuButton.get_popup()

var labels = [
	"A short label",
	"A moderately sized label",
	"An extremely long and unnecessarily lengthy label"
]

func _ready():
	popup.connect("index_pressed", self, "_on_index_pressed")

	var sc_short : = ShortCut.new()
	var sc_medium : = ShortCut.new()
	var sc_long : = ShortCut.new()

	var iek_short : = InputEventKey.new()
	var iek_medium : = InputEventKey.new()
	var iek_long : = InputEventKey.new()

	iek_short.set_scancode(KEY_S)

	iek_medium.set_scancode(KEY_M)
	iek_medium.control = true

	iek_long.set_scancode(KEY_L)
	iek_long.alt = true
	iek_long.control = true
	iek_long.meta = true
	iek_long.shift = true

	sc_short.set_shortcut(iek_short)
	sc_medium.set_shortcut(iek_medium)
	sc_long.set_shortcut(iek_long)

	var shortcuts : = [sc_short, sc_medium, sc_long]

	var label_index : = 0
	for i in range(3):
		for j in range(4):
			popup.add_item(labels[i])
		for k in range(4):
			if k == 0:
				label_index += 1
				continue
			popup.set_item_shortcut(label_index, shortcuts[k - 1], true)
			label_index += 1


func _on_index_pressed(index : int) -> void:
	popup.remove_item(index)
```
And here's a gif of the test:
Gif

Comment (akien-mga) [reply scene/gui/popup_menu.cpp]: Awesome work, thanks for the thorough check!

**Comment by akien-mga** [comment]
Note: to properly squash the commits you need to use `git rebase -i`, and then *force* push your changes to the remote branch to replace the previous commits, not merge. See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

**Comment by Awkor** [comment]
Sorry for not looking at it beforehand, I'll fix it ASAP

**Review by akien-mga — APPROVED** [review]
Thanks, that looks good!

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
It caused a regression in the editor, see #25900.

I'll revert for now, we can work on a better fix later on.

**Pull request merged**